plugins/modules/ec2_instance.py

```diff
@@ -2097,14 +2094,15 @@ def main():
     )
     client = module.client('ec2', retry_decorator=retry_decorator)
 
-    if module.params.get('filters') is None:
-        module.params['filters'] = build_filters()
+    filters = module.params.get('filters')
```

Comment:
I would write this like that: `filters = module.params.get('filters') or build_filters()`

Reply:
`or build_filters()` is not the same as `is None`, In theory `{}` (which later gets converted to []) is a valid filter, which the `or` variant would block.